Ruby 1.9 fails to install on Mint 19

```
$ rvm install ruby-1.9
...
Error running '__rvm_make -j4',
please read /home/nick/.rvm/log/1540442037_ruby-1.9.3-p551/make.log
There has been an error while running make. Halting the installation.
```
[make.log](https://github.com/rvm/rvm/files/2513290/make.log)

I'm on linux mint 19, so rvm also reports:

```
No binary rubies available for: mint/19/x86_64/ruby-2.0.0-p648.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
```

Looks like an openssl incompatibility:
```
In file included from ossl.h:57:0,
                 from ossl_ns_spki.c:11:
/usr/include/openssl/asn1_mac.h:10:2: error: #error "This file is obsolete; please update your software."
 #error "This file is obsolete; please update your software."
  ^~~~~
compiling callback-8.c
In file included from ossl.h:213:0,
                 from ossl_ns_spki.c:11:
openssl_missing.h:79:35: error: macro "EVP_MD_CTX_create" passed 1 arguments, but takes just 0
 EVP_MD_CTX *EVP_MD_CTX_create(void);
                                   ^
In file included from /usr/include/openssl/x509.h:23:0,
                 from /usr/include/openssl/x509v3.h:14,
                 from ossl.h:58,
                 from ossl_ns_spki.c:11:
openssl_missing.h:83:6: error: expected declaration specifiers or ‘...’ before ‘(’ token
 void EVP_MD_CTX_init(EVP_MD_CTX *ctx);
      ^
openssl_missing.h:91:6: error: expected declaration specifiers or ‘...’ before ‘(’ token
 void EVP_MD_CTX_destroy(EVP_MD_CTX *ctx);
      ^
Makefile:269: recipe for target 'ossl_ns_spki.o' failed
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,8 +14,7 @@
 * Fix trap restoration on Amazon Linux [\#4428](https://github.com/rvm/rvm/pull/4428)
 * Fix Amazon Linux 2 detection [\#4435](https://github.com/rvm/rvm/pull/4435)
 * Fix libssl dependency for Elementary 5.0 Juno [\#4448](https://github.com/rvm/rvm/pull/4448)
-* Fix libssl dependency for Mint 19 [\#4448](https://github.com/rvm/rvm/pull/4448)
-
+* Fix libssl dependency for Mint 19 [\#4482](https://github.com/rvm/rvm/pull/4482)
 
 #### Upgraded Ruby interpreters:
 * Add support for TruffleRuby 1.0.0-rc3 [\#4419](https://github.com/rvm/rvm/pull/4419), 1.0.0-rc5 [\#4440](https://github.com/rvm/rvm/pull/4440), 1.0.0-rc6 [\#4452](https://github.com/rvm/rvm/pull/4452), and 1.0.0-rc7 [\#4466](https://github.com/rvm/rvm/pull/4466).

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,8 @@
 * Fix trap restoration on Amazon Linux [\#4428](https://github.com/rvm/rvm/pull/4428)
 * Fix Amazon Linux 2 detection [\#4435](https://github.com/rvm/rvm/pull/4435)
 * Fix libssl dependency for Elementary 5.0 Juno [\#4448](https://github.com/rvm/rvm/pull/4448)
+* Fix libssl dependency for Mint 19 [\#4448](https://github.com/rvm/rvm/pull/4448)
+
 
 #### Upgraded Ruby interpreters:
 * Add support for TruffleRuby 1.0.0-rc3 [\#4419](https://github.com/rvm/rvm/pull/4419), 1.0.0-rc5 [\#4440](https://github.com/rvm/rvm/pull/4440), 1.0.0-rc6 [\#4452](https://github.com/rvm/rvm/pull/4452), and 1.0.0-rc7 [\#4466](https://github.com/rvm/rvm/pull/4466).
@@ -21,6 +23,7 @@
 
 #### Binaries:
 * Ubuntu 18.04 x64 binaries for Ruby 2.3.7, 2.4.4 and 2.5.1 [\#4438](https://github.com/rvm/rvm/issues/4438) 
+
 
 ## [1.29.4](https://github.com/rvm/rvm/tag/1.29.4)
 1 July 2018 - [Full Changelog](https://github.com/rvm/rvm/compare/1.29.3...1.29.4)

--- a/scripts/functions/requirements/mint
+++ b/scripts/functions/requirements/mint
@@ -14,6 +14,26 @@ requirements_mint_define_libreadline()
 
 requirements_mint_define_libssl()
 {
+  # Legacy libssl-dev required by older version of ruby has been renamed to libssl1.0-dev
+  # starting from Mint 19 (Tara)
+
+  case "$1" in
+    (ruby-2.3*|ruby-2.2*|ruby-2.1*|ruby-2.0*|ruby-1.9*)
+      if
+        __rvm_version_compare ${_system_version} -ge 19
+      then
+        undesired_check libssl-dev
+        requirements_check libssl1.0-dev
+      else
+        requirements_check libssl-dev
+      fi
+      ;;
+
+    (*)
+      requirements_check libssl-dev
+      ;;
+  esac
+
   requirements_check libssl-dev
 }
 

